lib/ansiblelint/errors.py

```diff
@@ -43,6 +44,7 @@ def __init__(
 
         self.message = message or getattr(rule, 'shortdesc', "")
         self.linenumber = linenumber
+        self.column = column
```

Comment:
Also the implementation of `__eq__` is flawed, it should compare `_hash_key`, not `__hash__()`. Probably makes sense to fix it in another PR though, since it has already been broken in the past.

Reply:
Comparing hashes is OK in this case since if keys are the same, the hashes will also be the same. It is a bit inefficient, but still current as far as the end-result goes.

Reply:
If you are assuming that hash collisions will not happen, it's correct. But Python's `hash` is not designed for strong collision resistance, so it should really not be used for this. Especially since most data structures which use hashes (like `set`, `dict`, etc.) assume that they can use equality to distinguish between objects which have the same hash value.

Reply:
Ugh, that is true. Highly unlikely, but still true.

Reply:
If you're using CPython on a 32-bit system, and manage to run ansible-lint on a large codebase with ~65536 errors, the chances of a collision is pretty high (birthday paradox). I agree that the chances of this happening are not that high, but if enough people run ansible-lint for enough many times, at some point it will happen. And since the code is more efficient without the hash, it's definitely worth getting rid of it :)